### PR TITLE
Added api validation

### LINKED ISSUE #16: Api request validator
# Description

We would like to validate requests in our apis to ensure we are ingesting valid data.

# Acceptence Criterea
-  add request validators for all DTOs and return descriptive error messages
- add file validators for all uploaded assets that checks file size and mime-type

### PR DESCRIPTION
**Pull request by aramikm**
# Description
Adding validation in API request.
closes #16

# Details
- Added route param and dto validation
- Added File size and mime type validation
- Added e2e tests for request validation

### CODE REVIEW

File: apps/api/src/api.controller.ts [hunk 1]
```diff
@@ -28,8 +40,23 @@ export class ApiController {
...
-  async assetUpload(@UploadedFiles() files: Array<Express.Multer.File>): Promise<UploadResponseDto> {
+  async assetUpload(
+    @UploadedFiles(
+      new ParseFilePipeBuilder()
+        .addFileTypeValidator({
+          fileType: DSNP_VALID_MIME_TYPES,
+        })
+        .addMaxSizeValidator({
+          // this is in bytes (2 GB)
+          maxSize: 2 * 1000 * 1000 * 1000,
```
Comment (aramikm) [inline apps/api/src/api.controller.ts:51]: Should be coming from config. I can either handle in this PR or in a future PR.

File: apps/api/src/api.controller.ts [hunk 2]
```diff
@@ -1,8 +1,20 @@
-import { Body, Controller, Delete, Get, HttpCode, HttpStatus, Logger, Param, Post, Put, UploadedFile, UploadedFiles, UseInterceptors } from '@nestjs/common';
+import { Body, Controller, Delete, Get, HttpCode, HttpStatus, Logger, Param, ParseFilePipeBuilder, Post, Put, UploadedFiles, UseInterceptors } from '@nestjs/common';
 import { v4 as uuidv4 } from 'uuid';
 import { FilesInterceptor } from '@nestjs/platform-express';
 import { ApiBody, ApiConsumes } from '@nestjs/swagger';
-import { BroadcastDto, ReactionDto, ReplyDto, UpdateDto, ProfileDto, AnnouncementResponseDto, FilesUploadDto, UploadResponseDto } from '../../../libs/common/src';
+import {
+  BroadcastDto,
```
Comment (saraswatpuneet) [inline apps/api/src/api.controller.ts:6]: nit: lint

File: apps/api/src/api.controller.ts (hunk 1 above)
Comment (saraswatpuneet) [reply apps/api/src/api.controller.ts:51]: Can we add a story or some bullet point to existing story to keep a track

**Review by saraswatpuneet — APPROVED** [review]
lgtm! minor comments here and there

File: apps/api/src/api.controller.ts (hunk 2 above)
Comment (aramikm) [reply apps/api/src/api.controller.ts:6]: This is done by linting, I think if the number of characters in imports length grows more than certain number it changes the structure to put everyone of them in a new line

File: apps/api/src/api.controller.ts (hunk 1 above)
Comment (aramikm) [reply apps/api/src/api.controller.ts:51]: created a new ticket https://github.com/AmplicaLabs/content-publishing-service/issues/20

**Pull request merged**